data-serving/data-service/src/model/case.ts

```diff
@@ -114,91 +105,126 @@ const pathogenSchema = new mongoose.Schema({
     },
-    sequenceSource: sourceSchema,
+    sequenceSource: {
+        type: sourceSchema,
+        validate: minimumSourceFieldsValidator,
+    },
     additionalInformation: String,
 });
 
-const pathogensSchema = new mongoose.Schema({
-    fields: [pathogenSchema],
-});
-
-const geometrySchema = new mongoose.Schema({
-    latitude: {
-        type: Number,
-        min: -90.0,
-        max: 90.0,
-    },
-    longitude: {
-        type: Number,
-        min: -180.0,
-        max: 180.0,
+const minimumLocationFieldsValidator = {
+    validator: function (location: Location) {
+        return (
+            !!location.country ||
+            !!location.administrativeAreaLevel1 ||
+            !!location.administrativeAreaLevel2 ||
+            !!location.locality
+        );
     },
-});
+    message:
+        'One of location.country, location.administrativeRegion1, ' +
+        'location.administrativeRegion2, or locality is required',
+};
 
 const locationSchema = new mongoose.Schema({
     id: String,
-
-    country: {
-        type: String,
-        required: 'Enter a country',
-    },
+    country: String,
     administrativeRegion1: String,
     administrativeRegion2: String,
-
     locality: String,
-    geometry: geometrySchema,
-});
-
-const dateRangeSchema = new mongoose.Schema({
-    start: Date,
-    end: Date,
+    geometry: {
+        latitude: {
+            type: Number,
+            min: -90.0,
+            max: 90.0,
+        },
+        longitude: {
+            type: Number,
+            min: -180.0,
+            max: 180.0,
+        },
+    },
 });
 
 const travelSchema = new mongoose.Schema({
-    location: locationSchema,
+    location: {
+        type: locationSchema,
+        validate: minimumLocationFieldsValidator,
+    },
     dates: dateRangeSchema,
     purpose: {
         type: String,
-        enum: Object.keys(TravelPurpose),
+        enum: Object.values(TravelPurpose),
     },
     additionalInformation: String,
 });
 
-const travelHistorySchema = new mongoose.Schema({
-    fields: [travelSchema],
-});
-
-const caseSchema = new mongoose.Schema({
-    chronicDisease: dictionaryValueSchema,
-    demographics: demographicsSchema,
-    eventSequence: {
-        type: eventSequenceSchema,
-        required: 'Enter an event sequence',
+const caseSchema = new mongoose.Schema(
+    {
+        chronicDisease: dictionaryValueSchema,
+        demographics: demographicsSchema,
+        events: {
+            type: [
+                {
+                    name: {
+                        type: String,
+                        required: 'Enter a name for the event',
+                    },
+                    date: dateRangeSchema,
+                },
+            ],
+            validate: {
+                validator: function (events: [Event]) {
+                    return events.some((e) => e.name == 'confirmed');
+                },
+                message: 'Must include a case confirmed event',
```

Comment:
nit: would be clearer to say 'an event with name equal to "case"'.

Reply:
done, thanks! (event with name "confirmed")